lighthouse-core/test/config/config-test.js

```diff
@@ -147,6 +147,74 @@ describe('Config', () => {
     })).toThrow('VRMLElements gatherer, required by audit missing-artifact-audit, was not found in config');
   });
 
+  // eslint-disable-next-line max-len
+  it('does not throw when an audit requests an optional artifact with no gatherer supplying it', async () => {
+    class DoesntNeedYourCrap extends Audit {
+      static get meta() {
+        return {
+          id: 'optional-artifact-audit',
+          title: 'none',
+          description: 'none',
+          requiredArtifacts: [
+            'URL', // base artifact
+            'ViewportDimensions', // from gatherer
+          ],
+          optionalArtifacts: [
+            'SourceMaps', // Not in the config.
+          ],
+        };
+      }
+
+      static audit() {}
+    }
+
+    // Shouldn't throw.
+    const config = new Config({
+      passes: [{
+        passName: 'defaultPass',
+        gatherers: [
+          'viewport-dimensions',
+        ],
+      }],
+      audits: [DoesntNeedYourCrap],
+    });
+    expect(config.passes[0].gatherers.map(g => g.path)).toEqual(['viewport-dimensions']);
+  });
+
+  it('does warn when an audit requests and receives an optional artifact', async () => {
+    class ButWillStillTakeYourCrap extends Audit {
+      static get meta() {
+        return {
+          id: 'optional-artifact-audit',
+          title: 'none',
+          description: 'none',
+          requiredArtifacts: [
+            'URL', // base artifact
+            'ViewportDimensions', // from gatherer
+          ],
+          optionalArtifacts: [
+            'SourceMaps', // Is in the config.
+          ],
+        };
+      }
+
+      static audit() {}
+    }
+
+    const config = new Config({
+      passes: [{
+        passName: 'defaultPass',
+        gatherers: [
+          'source-maps',
+          'viewport-dimensions',
+        ],
+      }],
+      audits: [ButWillStillTakeYourCrap],
+    });
+    expect(config.passes[0].gatherers.map(g => g.path))
```

Comment:
isn't this going to pass without your change? I think we need to add `onlyAudits: ['optional-artifact-audit']` to trigger filtering

Reply:
yea. consider this a regression test.

good point on triggering filtering.

Reply:
also need to have categories to filter ... so I just changed it to extend the default config.